src/common/components/cards/DeckSummary.tsx

```diff
@@ -1,31 +1,33 @@
 import * as React from 'react';
-import { arrayOf, bool, func, object } from 'prop-types';
 import Badge from 'material-ui/Badge';
 import Paper from '@material-ui/core/Paper';
 import FontIcon from 'material-ui/FontIcon';
 import { filter, sortBy } from 'lodash';
 
-import { TYPE_ROBOT, TYPE_EVENT, TYPE_STRUCTURE } from '../../constants.ts';
-import { groupCards } from '../../util/cards.ts';
-import { BUILTIN_FORMATS } from '../../util/formats.ts';
+import * as w from '../../types';
+import { TYPE_ROBOT, TYPE_EVENT, TYPE_STRUCTURE } from '../../constants';
+import { groupCards } from '../../util/cards';
+import { BUILTIN_FORMATS, SetFormat } from '../../util/formats';
 import ButtonInRow from '../ButtonInRow';
 import Tooltip from '../Tooltip';
-import CardTooltip from '../card/CardTooltip.tsx';
+import CardTooltip from '../card/CardTooltip';
 import MustBeLoggedIn from '../users/MustBeLoggedIn';
 
-export default class DeckSummary extends React.Component {
-  static propTypes = {
-    cards: arrayOf(object),
-    deck: object,
-    loggedIn: bool,
-
-    onDelete: func,
-    onDuplicate: func,
-    onEdit: func,
-    onTry: func
-  };
+import { CardWithCount } from './types';
+
+interface DeckSummaryProps {
+  cards: w.CardInStore[]
+  deck: w.DeckInStore
+  set?: w.Set
+  loggedIn: boolean
+  onDelete: (deckId: string) => void
+  onDuplicate: (deckId: string) => void
+  onEdit: (deckId: string) => void
+  onTry: (deck: w.DeckInStore) => void
+}
 
-  get styles() {
+export default class DeckSummary extends React.Component<DeckSummaryProps> {
+  get styles(): Record<string, Record<string, string | number>> {
```

Comment:
Should `Record<string, Record<string, string | number>>` be an interface or type alias since it'll be used in so many files?

Reply:
Or `Record<string, CSSProperties>` as you use that in other places